[CI] ROS: Add friendly names; add TSID on Rolling as downstream test

wxmerkt opened the pull request:
Adding a test to build TSID with ProxQP as a downstream test for ROS packaging on Rolling

jcarpent reviewed (approved):
Very nice @wxmerkt. 
Thanks a lot

wxmerkt commented:
@jcarpent The downstream TSID test is now working. I also updated the package.xml and added catkin/ament integration as with Pinocchio etc. - this should be good to merge. Could you please create another patch release with these updates? Once we get the release repository happy to run the release for Rolling/Humble
